File: app/src/main/java/com/pietrantuono/podcasts/addpodcast/repository/repository/Repository.java
package com.pietrantuono.podcasts.addpodcast.repository.repository;

import com.pietrantuono.podcasts.addpodcast.model.pojos.Podcast;

import org.jetbrains.annotations.NotNull;

import java.util.List;

import rx.Observable;
import rx.Observer;

public interface Repository {
    Observable<Boolean> getIfSubscribed(Podcast trackId);

    Observable<List<Podcast>> getSubscribedPodcasts(Observer<List<Podcast>> observer);

    void subscribeUnsubscribeToPodcast(Podcast podcast);

    @NotNull Observable<? extends Podcast> getPodcastById(int trackId);
}

File: app/src/main/java/com/pietrantuono/podcasts/repository/repository/Repository.kt
package com.pietrantuono.podcasts.repository.repository

import com.pietrantuono.podcasts.addpodcast.model.pojos.Podcast

import rx.Observable
import rx.Observer

interface Repository {
    fun getIfSubscribed(trackId: Podcast): Observable<Boolean>
    fun getSubscribedPodcasts(observer: Observer<List<Podcast>>): Observable<List<Podcast>>
    fun subscribeUnsubscribeToPodcast(podcast: Podcast)
    fun getPodcastById(trackId: Int): Observable<out Podcast>
}
